src/main/kotlin/org/rust/ide/inspections/UnnecessaryParenthesisInspection.kt

```diff
@@ -0,0 +1,35 @@
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.LocalQuickFixBase
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.rust.lang.core.psi.RustIfExpr
+import org.rust.lang.core.psi.RustVisitor
+import org.rust.lang.core.psi.impl.RustParenExprImpl
+
+/**
+ * Created by zjh on 16/3/8.
+ */
+class UnnecessaryParenthesisInspection : RustLocalInspectionTool() {
+    override fun getDisplayName() = "Unnecessary parenthesis"
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor =
+        object : RustVisitor() {
+            override fun visitIfExpr(expr: RustIfExpr) {
+                val exprText = expr.expr?.text
+                if (exprText?.startsWith("(")!! && exprText?.endsWith(")")!!) {
```

Comment:
This check is better written as `expr.expr is RustParenExpr`. In general it is much better to operate on PSI structure, rather then on raw text. In particular, this check will be wrong for this example

```Rust
if (true) or (false) {}
```

Also you should avoid `!!` at all costs. If something is nullable it is usually nullable for a reason, and thus `!!` is just a lazily evaluated NPE :)

In this particular casse, something like

```
val cond = expr?.expr ?: return
``

can be used. Here are some docs on dealing with nullability in Kotlin: https://kotlinlang.org/docs/reference/null-safety.html